bionic/persistence.py

```diff
@@ -754,43 +750,35 @@ class LocalFilesystem:
     to local disk.
     """
 
-    def __init__(self, inventory_dir, tmp_dir):
+    def __init__(self, inventory_dir):
         self.root_url = url_from_path(inventory_dir)
-        self.tmp_root_path = tmp_dir
+        self._fs = fsspec.filesystem("file")
 
     def exists(self, url):
-        return path_from_url(url).exists()
+        return self._fs.exists(unquote(url))
 
     def search(self, url_prefix):
-        path_prefix = path_from_url(url_prefix)
-        if not path_prefix.is_dir():
-            return []
-
-        return [
-            url_from_path(path_prefix / sub_path)
-            for sub_path in path_prefix.glob("**/*")
-        ]
+        # This endpoint is a glob **/* search.
+        glob_url = unquote(url_prefix) + "**/*"
+        return ["file://" + url for url in self._fs.glob(glob_url)]
```

Comment:
Wondering, we cannot use `url_from_path` here?

Reply:
We'll have to convert urls to paths first but that should work too!